jena-arq/src/main/java/org/apache/jena/riot/other/G.java

```diff
@@ -0,0 +1,593 @@
+
+    /** Return the subject of a triple, or null if the triple is null. */
+    public static Node subject(Triple triple) {
+        return triple == null ? null : triple.getSubject();
+    }
+
+    /** Return the predicate of a triple, or null if the triple is null. */
+    public static Node predicate(Triple triple) {
+        return triple == null ? null : triple.getPredicate();
+    }
+
+    /** Return the object of a triple, or null if the triple is null. */
+    public static Node object(Triple triple) {
+        return triple == null ? null : triple.getObject();
+    }
+
+    // ---- Node filter tests.
+    public static boolean isURI(Node n)         { return n != null && n.isURI(); }
+    public static boolean isBlank(Node n)       { return n != null && n.isBlank(); }
+    public static boolean isLiteral(Node n)     { return n != null && n.isLiteral(); }
+    public static boolean isResource(Node n)    { return n != null && (n.isURI()||n.isBlank()); }
+
+    /** Convert null to Node.ANY */
+    public static Node nullAsAny(Node x) { return nullAsDft(x, Node.ANY) ; }
+
+    /** Convert null to some default Node */
+    public static Node nullAsDft(Node x, Node dft) { return x==null ? dft : x ; }
+
+    /** Does the graph match the s/p/o pattern? */ 
+    public static boolean contains(Graph g, Node s, Node p, Node o) {
+        return g.contains(s, p, o);
+    }
+    
+    /** Does the graph use the node anywhere as a subject, predicate or object? */
+    public static boolean containsNode(Graph graph, Node node) {
+        return GraphUtil.containsNode(graph, node);
+//        return
+//            contains(graph, node, Node.ANY, Node.ANY) ||
+//            contains(graph, Node.ANY, Node.ANY, node) ||
+//            contains(graph, Node.ANY, node, Node.ANY) ;
+    }
+    
+    /** Test whether the node has the type or is rdfs:subclassOf. */
+    public static boolean isOfType(Graph graph, Node x, Node type) {
+        Objects.requireNonNull(x, "Subject");
+        Objects.requireNonNull(type, "Type");
+        List<Node> allClasses = listSubClasses(graph, type);
+        for ( Node c : allClasses ) {
+            if ( hasType(graph, x, c) )
+                return true;
+        }
+        return false;
+    }
+
+    /** Does the node x have the given type (non-RDFS - no rdfs:subclassOf considered)? */ 
+    public static boolean hasType(Graph graph, Node x, Node type) {
+        Objects.requireNonNull(x, "Subject");
+        Objects.requireNonNull(type, "Type");
+        return contains(graph, x, NodeConst.nodeRDFType, type);
+    }
+
+    //---- get/list/iter
+
+    /** Does node {@code s} have property {@code p} in graph {@code g}? */
+    public static boolean hasProperty(Graph g, Node s, Node p) {
+        return g.contains(s, p, null);
+    }
+
+    /** Contains exactly one. */
+    public static boolean containsOne(Graph g, Node s, Node p, Node o) {
+        return g.contains(s, p, o) && Iter.count(g.find(s,p,null)) == 1;
```

Comment:
Could be more efficient without counting all triples, e.g. 
```
try (ExtendedIterator<Triple> it = g.find(s, p, null) {
  if (!it.hasNext()) {
     return false;
  }
  it.next();
  return !it.hasNext();
}
```

Reply:
Agreed.

Reply:
Actually, try-with-resources wouldn't work here as ClosableIterator doesn't implement AutoCloseable. It would be quite convenient and shouldn't break anything. `public interface ClosableIterator<T> extends Iterator<T>,  AutoCloseable`

Reply:
See https://issues.apache.org/jira/browse/JENA-1956...

Reply:
Ah great!